oracle/oracle.go

```diff
@@ -0,0 +1,235 @@
+package oracle
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/libopenstorage/cloudops"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/containerengine"
+	"github.com/oracle/oci-go-sdk/v65/core"
+)
+
+const (
+	metadataAPIEndpoint           = "http://169.254.169.254/opc/v2/instance/"
```

Comment:
I see that you are using the Metadata version 2 directly. We might want to evaluate a few things first.
Not all images will support v2: https://docs.oracle.com/en-us/iaas/Content/Compute/Tasks/gettingmetadata.htm#upgrading-v2__supported-images
We might also want to support v1 if we end up with one of the unsupported images.

Reply:
Difference between v2 and v1 is, v2 requires additional request header called `Authorization: Bearer Oracle` and a different endpoint. JSON Response returned by both v1 and v2 is same. Execution of tests passed when I changed the `metadataAPIEndpoint ` to  `"http://169.254.169.254/opc/v2/instance/"`

May be lets perform metadata lookup to v1 endpoint first and if it fails then use v2.

Reply:
Sure.

Reply:
Using HTTP 404 NOT FOUND response to detect whether endpoint is enabled is more easier than to use GetInstance() API since 2nd approach requires compute client which again depends on certain things like region and tenancy ID that we are planning to get from metadata service itself than from user input.